jsonschema-circe/src/main/scala/sttp/apispec/internal/JsonSchemaCirceEncoders.scala

```diff
@@ -13,161 +13,142 @@ trait JsonSchemaCirceEncoders {
+          "minContains" := s.minContains,
+          "prefixItems" := s.prefixItems,
+          "items" := s.items,
+          "contains" := s.contains,
+          "unevaluatedItems" := s.unevaluatedItems,
+          "maxProperties" := s.maxProperties,
+          "minProperties" := s.minProperties,
+          "required" := s.required,
+          "dependentRequired" := s.dependentRequired,
+          "discriminator" := s.discriminator,
+          "properties" := s.properties,
+          "patternProperties" := s.patternProperties,
+          "additionalProperties" := s.additionalProperties,
+          "propertyNames" := s.propertyNames,
+          "unevaluatedProperties" := s.unevaluatedProperties,
+          "format" := s.format,
+          "title" := s.title,
+          "description" := s.description,
+          "default" := s.default,
+          "deprecated" := s.deprecated,
+          "readOnly" := s.readOnly,
+          "writeOnly" := s.writeOnly
+        ) ++ exampleFields ++ Vector(
+          "externalDocs" := s.externalDocs,
+          "extensions" := s.extensions
+        )
       )
-
-      // nullable ref types in openapi 3.0
-      // See: https://stackoverflow.com/questions/40920441/how-to-specify-a-property-can-be-null-or-a-reference-with-swagger
-      if (needsNullableWrapper) JsonObject("allOf" := Json.arr(result.asJson), "nullable" := true)
-      else result
     }
     .mapJsonObject(expandExtensions)
 
   // note: these are strict val-s, order matters!
   implicit val extensionValue: Encoder[ExtensionValue] =
     Encoder.instance(e => parse(e.value).getOrElse(Json.fromString(e.value)))
 
-  implicit val encoderExampleSingleValue: Encoder[ExampleSingleValue] = {
-    case ExampleSingleValue(value: String)     => parse(value).getOrElse(Json.fromString(value))
-    case ExampleSingleValue(value: Int)        => Json.fromInt(value)
-    case ExampleSingleValue(value: Long)       => Json.fromLong(value)
-    case ExampleSingleValue(value: Float)      => Json.fromFloatOrString(value)
-    case ExampleSingleValue(value: Double)     => Json.fromDoubleOrString(value)
-    case ExampleSingleValue(value: Boolean)    => Json.fromBoolean(value)
-    case ExampleSingleValue(value: BigDecimal) => Json.fromBigDecimal(value)
-    case ExampleSingleValue(value: BigInt)     => Json.fromBigInt(value)
-    case ExampleSingleValue(null)              => Json.Null
-    case ExampleSingleValue(value)             => Json.fromString(value.toString)
-  }
-
-  implicit val encoderMultipleExampleValue: Encoder[ExampleMultipleValue] = { e =>
-    Json.arr(e.values.map(v => encoderExampleSingleValue(ExampleSingleValue(v))): _*)
-  }
-
   implicit val encoderExampleValue: Encoder[ExampleValue] = {
-    case e: ExampleMultipleValue => encoderMultipleExampleValue.apply(e)
-    case e: ExampleSingleValue   => encoderExampleSingleValue.apply(e)
+    case ExampleValue(value: String)     => parse(value).getOrElse(Json.fromString(value))
```

Comment:
do we somehow retain the possibility for examples with multiple values? e.g. arrays

Reply:
`ExampleValue` wraps an `Any` and this encoder indeed does not seem to understand arrays. In my opinion it should accept a `Json` instead of `Any` - but this would require a premature dependency on `circe`. Therefore the right way to handle this is IMO by accepting an arbitrary JSON string.

Reply:
Here's how we use it in tapir: https://github.com/softwaremill/tapir/blob/604bd9da5a5374235135f1d25afef3e5a3c44346/docs/apispec-docs/src/main/scala/sttp/tapir/docs/apispec/package.scala#L45

Over there, we also still don't have a specific serialization format (json). Though maybe ... it's already encoded by the codec into a string represneting a json array, so maybe this is never called. Something to verify :)

Reply:
Wait, so `ExampleMultipleValue` was used to express something like this?

```
"example": [ /* multiple examples here */]
```

As far as I understand it, this is not valid in a schema. Or - it is technically valid - but expresses something else that was intended: a single example value that is an array, rather than multiple example values.

Or was that supposed to be some kind of a hack for lack of support of multiple examples in OAS 3.0?

Reply:
No, it was used to express a single example, which is a collection of "example-items that we know how to serialise".

It is a special-case (we only go one level deep when inspecting example values), but maybe there are some use-cases in tapir tests.

Reply:
What do you mean by "know how to serialize"?
All example values ultimately must be serializable to JSON (even if YAML or something else is the final format), so I don't yet see a reason to introduce this special case. Although I'm probably still missing some context.

Reply:
I've redesigned `ExampleValue` and created a [parallel PR in tapir](https://github.com/softwaremill/tapir/pull/3677) to track the consequences of this one.

Reply:
Well an example doesn't have to be JSON - it can be anything to what the codec serialises the high-level value to. You can embed an XML into a string just as well :) 

JSON is kind of a special case, as if the value is valid json, we can parse it and represent "in-line" (both in the YAML and JSON representation of OpenAPI), instead of embedding as a string

Reply:
Also, this specific line you commented on:

```scala
case ExampleValue(value: String)     => parse(value).getOrElse(Json.fromString(value))
```

makes me _extremely_ uncomfortable: what if the `String` passed to `ExampleValue` just accidentally happens to be a valid JSON, even though someone's intention was to pass a plain string? Things like that tend to cause some of the worst bugs.

Reply:
I know it's not ideal, but it works and nobody so far complained ;) Unfortunately at this point we simply don't know what's the original serialisation format. But I'm open to better solutions :)

Reply:
I reverted changes to `ExampleValue` as it's more complicated than expected and not in the scope of the ultimate goal (https://github.com/softwaremill/tapir/issues/3645)